### DISCUSSION TITLE
Reload i18n resources

### DISCUSSION THREAD

**Discussion by zabihexes** [ivanhofer/typesafe-i18n#566] [Q&A] [answered]
Is there any way for me to reload i18next resources while using `typesafeI18nObject`?
Normally I would use `i18next.reloadResources`, but I do not really have access to that with `typesafeI18nObject`.

**Comment by zabihexes** [comment]
I could call `typesafeI18nObject` again and replace the whole thing, but that does not seem like a good solution.

**Comment by zabihexes** [comment]
Same issue if I use the `i18n` function btw.

**Comment by ivanhofer** [comment]
What do you exactly mean with "reloading"?
Does your dictionary change while you are using the application?

**Reply by zabihexes** [reply]
Yes, the dictionary can change if I manually trigger a reload of it. It is a common practice when developing discord bots. (To be short, there are good reasons for it.)

I could just reassign the variable(`LL`, I think you named it like that in examples.) with output of `typesafeI18nObject`(calling the function again) again on each reload, but I do not think that is ideal.

**Reply by zabihexes** [reply]
Reloading is basically importing a file again and assigning the output of it to some variable. Then I would pass that variable to i18n.

It is a similar idea to HMR(hot module reload)  although it works differently.

**Reply by ivanhofer** [reply]
For me this sounds like the same process as switching to another locale. There you first load the translations from disk via `loadLocaleAsync` and then call the `setLocale` function from the specific adapter (like [here](https://github.com/ivanhofer/typesafe-i18n/blob/main/packages/adapter-react/src/index.tsx#L58)). There also a new instance of `i18nObject` get's created.

**Comment by zabihexes** [comment, marked as answer]
Got it working. Had to copy whole `i18n` function (https://github.com/ivanhofer/typesafe-i18n/blob/main/packages/runtime/src/util.instance.mts#L16) so I can access the `cache` variable myself. Rest was just replacing locales in the cache with my reloaded one.

**Reply by ivanhofer** [reply]
Great that you have found a solution that works for you  :)
